app/queries/display_ads/filtered_ads_query.rb

```diff
@@ -59,33 +60,26 @@ def authenticated_ads(display_auth_audience)
       @filtered_display_ads.where(display_to: display_auth_audience)
     end
 
-    def community_or_in_house_ads
-      @filtered_display_ads.where(
-        "(type_of = :in_house) OR
-         (type_of = :community AND organization_id = :organization_id) OR
-	       (type_of = :external AND organization_id != :organization_id)",
-        DisplayAd.type_ofs.merge({ organization_id: @organization_id }),
-      )
-    end
-
-    # Business Logic Context:
-    # We are always showing more of the good stuff — but we are also always testing the system to give any a chance to
-    # rise to the top. 1 out of every 8 times we show an ad (12.5%), it is totally random. This gives "not yet
-    # evaluated" stuff a chance to get some engagement and start showing up more. If it doesn't get engagement, it
-    # stays in this area.
-
-    # Ads that get engagement have a higher "success rate", and among this category, we sample from the top 15 that
-    # meet that criteria. Within those 15 top "success rates" likely to be clicked, there is a weighting towards the
-    # top ranked outcome as well, and a steady decline over the next 15 — that's because it's not "Here are the top 15
-    # pick one randomly", it is actually "Let's cut off the query at a random limit between 1 and 15 and sample from
-    # that". So basically the "limit" logic will result in 15 sets, and then we sample randomly from there. The
-    # "first ranked" ad will show up in all 15 sets, where as 15 will only show in 1 of the 15.
-    def sample_ads
-      if rand(8) == 1
-        @filtered_display_ads.sample
-      else
-        @filtered_display_ads.limit(rand(1..15)).sample
-      end
+    def type_matched_ads
+      # Always match in-house-type ads
+      in_house = "(type_of = :in_house)"
+
+      # If this is an article that belongs to an organization, we might show community-type ads
+      community = if @organization_id
+                    "(type_of = :community AND organization_id = :organization_id)"
+                  end
+
+      # If the article's author permits adjacent sponsors, we might show an external-type ad
+      # *if* the organization_id doesn't match the current article's organization id
+      external = if @permit_adjacent_sponsors && @organization_id
+                   "(type_of = :external AND organization_id != :organization_id)"
+                 elsif @permit_adjacent_sponsors
+                   "(type_of = :external)"
+                 end
```

Comment:
`cc` @Ridhwana @lboogie04 I'm not sure if this change is _really_ necessary, maybe it's fine, but I think database generally treat `NULL` in a way that makes it work strangely if it winds up in a `!= null` clause. ("NULL is a placeholder, not a value" is how I learned to think about it, but I'm no longer confident that any/all databases treat this the same way.)

Reply:
@jaw6 I was doing some testing this morning and this is definitely necessary, otherwise it was yielding the results whereby the ad was not showing up articles where the article `organization_id` was nil.